### ISSUE TITLE
Muted TypeScript error / handling onBeforeOpen + onClose events

### ISSUE THREAD

**Issue by hw-df** [jonataswalker/ol-contextmenu#233]
The included example uses a workaround to prevent TypeScript error:
https://github.com/jonataswalker/ol-contextmenu/blob/d5f2d6a0d9be7f8ac6ef3e643152a6803734c4be/examples/my-project-with-ol-contextmenu/src/main.ts#L106-L107
But this indicates code problems and is not allowed in some environments (like mine), so this shouldn't be the final solution.

The ts error here is:

```
TS2769: No overload matches this call.

  Overload 1 of 3, '(type: EventTypes, listener: (event: BaseEvent) => unknown): EventsKey', gave the following error.
    Argument of type '"beforeopen"' is not assignable to parameter of type 'EventTypes'.

  Overload 2 of 3, '(type: "propertychange", listener: (event: ObjectEvent) => unknown): EventsKey', gave the following error.
    Argument of type '"beforeopen"' is not assignable to parameter of type '"propertychange"'.

  Overload 3 of 3, '(type: (EventTypes | "propertychange")[], listener: (event: Event | BaseEvent) => unknown): EventsKey[]', gave the following error.

    Argument of type 'string' is not assignable to parameter of type '(EventTypes | "propertychange")[]'.

```

I stumbled upon this OL issue: https://github.com/openlayers/openlayers/issues/12512
Maybe it´s not possible to extend the ol/Obervable "on"-listener at the moment? It`s working with workaround, but that might change at any time.

An alternative solution could be to let define custom callback handler in options. Like:
```
const contextmenu = new ContextMenu({
    width: 200,
    defaultItems: true,
    items,
    beforeOpenHandler: myBeforeOpenHandler
});
```

**Comment by jonataswalker** [comment]
Hi there @hw-df 

Are you able to confirm if https://github.com/jonataswalker/ol-contextmenu/pull/238 (published as `v5.1.1`) fixed this issue?

**Comment by hw-df** [comment]
Yes, that fixed this issue. Thank you!
